### PR TITLE
Update SystemInfoHelper.cs

### PR DESCRIPTION
**Pull request by ABCDCreeper**
优化了Java搜索

### CODE REVIEW

File: ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs [hunk 1]
```diff
@@ -38,64 +43,30 @@ public static MemoryInfo GetMemoryUsage()
     public static async IAsyncEnumerable<string> FindJava(bool fullSearch = false)
     {
         var result = new HashSet<string>();
+        var DiskName = getDisk();
```
Comment (CodingEric) [inline ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs:46]: The code style here should be consistent with other parts of the projects.

File: ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs [hunk 2]
```diff
@@ -38,64 +43,30 @@ public static MemoryInfo GetMemoryUsage()
...
-                    result.Add(FullPath);
+                DirectoryInfo TheFolder = new DirectoryInfo($"{JP}\\Program Files\\Java");
+                foreach (DirectoryInfo NextFolder in TheFolder.GetDirectories())
+                {
+                    string FullPath = JP+"\\Program Files\\Java\\" + NextFolder.Name + "\\bin\\javaw.exe";
+                    if (File.Exists(FullPath))
+                        result.Add(FullPath);
+                }
             }
+            catch { }
```
Comment (CodingEric) [inline ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs:64]: Unnecessary trying & catching, you had better replace it with logical judgements which are more explicit.

File: ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs [hunk 3]
```diff
@@ -143,4 +114,19 @@ static string FindJavaUsingEnvironmentVariable()
             return string.Empty;
         }
     }
+
+
+    public static List<string> getDisk()
```
Comment (CodingEric) [inline ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs:119]: Repeated and inefficient implementation. Please refer to DeepJavaSearcher.cs .

File: ProjBobcat/Example/Program.cs [hunk 4]
```diff
@@ -21,13 +21,18 @@ private static async Task Main(string[] args)
...
-
+            var Disk = SystemInfoHelper.getDisk();
+            var DiskResult = new List<string>();
             await foreach (var java in jl)
             {
                 Console.WriteLine($"搜索到的 Java - {java}");
                 javaResult.Add(java);
             }
-
+            foreach (var disk in Disk)
```
Comment (CodingEric) [inline ProjBobcat/Example/Program.cs:31]: Unnecessary.

**Review by CodingEric — CHANGES_REQUESTED** [review]
We can't merge this PR before the problems being addressed.

File: ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Management;
```
Comment (CodingEric) [inline ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs:4]: Note that .NET Core does not support System.Management

**Comment by ABCDCreeper** [comment]
Fixed

**Pull request closed without merging**